src/test/functional/queryServices.test/tags.integration.ts

```diff
@@ -444,69 +441,183 @@ describe('tags query tests - happy path', () => {
         }
       }
     `;
+    beforeAll(async () => {
+      getTagsByNameSpy = sinon.spy(TagDataService.prototype, 'getTagsByName');
+      await db('readitla_b.item_grouping').insert([
+        {
+          resolved_id: 1,
+          grouping_id: 10631688,
+          source_score: 1.0,
+        },
+        {
+          resolved_id: 1,
+          grouping_id: 10631689,
+          source_score: 2.0,
+        },
+      ]);
+
+      await db('suggested_tags_user_grouping_tags').insert([
+        {
+          user_id: 1,
+          grouping_id: 10631688,
+          tag: 'zebra',
+          weighted_count: 0.9211,
+          count: 2,
+        },
+        {
+          user_id: 1,
+          grouping_id: 10631689,
+          tag: 'travel',
+          weighted_count: 0.9211,
+          count: 2,
+        },
+      ]);
+
+      await db('suggested_tags_user_groupings').insert([
+        {
+          user_id: 1,
+          grouping_id: 10631688,
+          weighted_count: 1.6461,
+          count: 1,
+        },
+        {
+          user_id: 1,
+          grouping_id: 10631689,
+          weighted_count: 1.6461,
+          count: 1,
+        },
+      ]);
+    });
+    afterAll(async () => {
+      await Promise.all(
+        [
+          'suggested_tags_user_groupings',
+          'suggested_tags_user_grouping_tags',
+          'readitla_b.item_grouping',
+        ].map((table) => db(table).truncate())
+      );
+      getTagsByNameSpy.restore();
+    });
 
-    const res = await server.executeOperation({
-      query: GET_SUGGESTED_TAGS_FOR_SAVED_ITEM,
-      variables,
+    afterEach(() => {
+      getTagsByNameSpy.resetHistory();
+    });
+
+    it('return list of SuggestedTags and paginated savedItems for SavedItem, ordered by score descending', async () => {
+      const res = await server.executeOperation({
+        query: GET_SUGGESTED_TAGS_FOR_SAVED_ITEM,
+        variables,
+      });
+      const expectedTags = [
+        {
+          id: Buffer.from('travel').toString('base64'),
+          name: 'travel',
```

Comment:
i might be missing context, in this test case -  both `zebra` and `travel` already would show up as tags as they have `item_tags` association with item_id: 1,
how does suggested tags differ here?

Reply:
how i understood this: say if the user has 1000 tags in `item_tags` table - then we would return the closest few tags as suggested but it would be a long list to test here